ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestCompactionQueryBuilderForMmCompaction.java

```diff
@@ -0,0 +1,472 @@
+    CompactionQueryBuilder queryBuilder = getMmMajorCompactionQueryBuilderForCreate();
+    Table sourceTable = createSourceTable();
+    queryBuilder.setSourceTab(sourceTable);
+    StorageDescriptor storageDescriptor = createStorageDescriptor();
+    queryBuilder.setStorageDescriptor(storageDescriptor);
+    String query = queryBuilder.build();
+    String expectedQuery =
+        "CREATE temporary external table comp_test_result_table(`column_1` string,`column_2` int,`column_3` boolean)  ROW FORMAT SERDE '/some/test/serialization_lib'WITH SERDEPROPERTIES ( \n" + "  'test_param_1'='test_value', \n" + "  'test_param_2'='test_value')STORED AS INPUTFORMAT 'some.test.InputFormat' OUTPUTFORMAT 'some.test.OutputFormat' TBLPROPERTIES ('transactional'='false')";
+    Assert.assertEquals(expectedQuery, query);
+  }
+
+  @Test
+  public void testMajorCompactionCreateWithSkewedByClause() throws HiveException {
+    CompactionQueryBuilder queryBuilder = getMmMajorCompactionQueryBuilderForCreate();
+    Table sourceTable = createSourceTable();
+    StorageDescriptor storageDescriptor = sourceTable.getSd();
+    SkewedInfo skewedInfo = new SkewedInfo();
+    skewedInfo.addToSkewedColNames("column_1");
+    List<String> skewedColValues = new ArrayList<>();
+    skewedColValues.add("value1");
+    skewedColValues.add("value2");
+    skewedColValues.add("value3");
+    skewedInfo.addToSkewedColValues(skewedColValues);
+    storageDescriptor.setSkewedInfo(skewedInfo);
+    storageDescriptor.setStoredAsSubDirectories(true);
+    sourceTable.setSd(storageDescriptor);
+    queryBuilder.setSourceTab(sourceTable);
+    String query = queryBuilder.build();
+    String expectedQuery =
+        "CREATE temporary external table comp_test_result_table(`column_1` string,`column_2` int,`column_3` boolean)  SKEWED BY (column_1) ON ('value1','value2','value3')) STORED AS DIRECTORIES TBLPROPERTIES ('transactional'='false')";
+    Assert.assertEquals(expectedQuery, query);
+  }
+
+  @Test
+  public void testMajorCompactionCreateWithNonNativeTable() throws HiveException {
+    CompactionQueryBuilder queryBuilder = getMmMajorCompactionQueryBuilderForCreate();
+    Table sourceTable = createSourceTable();
+    Map<String, String> parameters = new HashMap<>();
+    parameters.put("storage_handler", "test_storage_handler");
+    sourceTable.setParameters(parameters);
+    queryBuilder.setSourceTab(sourceTable);
+    StorageDescriptor storageDescriptor = createStorageDescriptor();
+    queryBuilder.setStorageDescriptor(storageDescriptor);
+    Exception exception = Assert.assertThrows(RuntimeException.class, queryBuilder::build);
+    String expectedMessage =
+        "Table comp_test_source_tablehas a storage handler (test_storage_handler). Failing compaction for this non-native table.";
+    String actualMessage = exception.getMessage();
+    Assert.assertTrue(actualMessage.contains(expectedMessage));
```

Comment:
Can we change this logic to something like
```
    String expectedMessage =
        "Table comp_test_source_tablehas a storage handler (test_storage_handler). Failing compaction for this non-native table.";
    Assert.assertThrows(expectedMessage, RuntimeException.class, queryBuilder::build);
```

& in other tests as well, if we are doing something similar

Reply:
Sure, fixed it.